scene/gui/file_dialog.cpp

```diff
@@ -174,6 +176,7 @@ void FileDialog::_native_dialog_cb_with_options(bool p_ok, const Vector<String>
 	file->set_text(f);
 	dir->set_text(f.get_base_dir());
 	filter->select(p_filter);
+	update_dir();
```

Comment:
```suggestion
	_change_dir(f.get_base_dir());
```

`current_dir` would just be reset to the same value when using `update_dir()`. Calling `_change_dir()` properly updates the `curent_dir` and also calls `update_dir`.

But this made me discover a new bug: since you're changing this *after* the `*_selected` signals are emitted, you can't retrieve the new `current_dir` value unless you waited a frame in the signal handler.

This is *not* a problem for embedded FileDialogs, only native ones. Do you think the fix is as easy as moving this update-logic above the signal emission? Or should I open a new bug ticket for it instead?

Reply:
Yes, probably moving it will work fine.

Reply:
I tried moving it and it works as expected.
Not sure if you saw but I updated my above comment since I found 2 more semi-related bugs, regarding the values of `current_dir` and `current_path` after selecting a file/dir in dialog.

Reply:
I fixed it like this, feel free to cherry-pick: https://github.com/RedMser/godot/commit/bf1d24bfdfa7ee862d33f4da715e66b1ec323fc9

This also fixes embedded file dialogs, which used to return paths like `/Users/Public/...` for `current_dir` and `current_path`, now also returning the drive letter as expected.

Reply:
Not sure what changing `dir_access->get_current_dir()` to `full_path` should do, it should always be the same value.

Reply:
> Yes I guess the EditorFileDialog change is not strictly needed then.

Missed that it is different in `FileDialog` (why do we have two almost identical dialogs?), this make sense for the `FileDialog`. Updated PR.